File: A02/Python/P02/a.-FuerzaBruta1.py
import time

def format_time(t):
    time_formatted = ""
    hour = minute = second = milisecond = int(0)

    hour = int(t//3600);
    t = t - (hour*3600)
    minute = int(t//60)
    t = t - (minute*60)
    second = int(t//1)
    t = t - second
    t = int(t*10000)
    milisecond = t
    
    time_formatted += (("0" + str(hour)) if hour < 10 else str(hour)) + ':'
    time_formatted += (("0" + str(minute)) if minute < 10 else str(minute)) + ':'
    time_formatted += (("0" + str(second)) if second < 10 else str(second)) + '.'
    time_formatted += '{:0>4}'.format(milisecond)

    return time_formatted


def is_prime(num):
    res = True

    for i in range(2, num):
        if num % i == 0: res = False

    return res
    

def main():
    primes = 0
    exec_time = ""
    t = 0.0
    
    t = time.time()
    while True:
        number = int(input())
        if number == -1: break
        if is_prime(number): primes += 1
    t = time.time() - t
    exec_time = format_time(t);

    print("Total:", primes)
    print("Tiempo:", exec_time)

if __name__ == "__main__":
    main()

File: A02/C++/P03/a.-FuerzaBruta1.cpp
#include <iostream>
#include <iomanip>
#include <sstream>
#include <string>
#include <ctime>
#define MIN 3221225472
#define MAX 3221226472
using namespace std;

string formatTime(long double t) {
	ostringstream time;
	int hour, min, sec, milisec;
	
	hour = t/3600;
	t = t - (hour*3600);
	min = t/60;
	t = t - (min*60);
	sec = t;
	t = t - (sec/1);
	milisec = t*10000;

	time << setfill('0') <<
		setw(2) << hour << ':' <<
		setw(2) << min << ':' <<
		setw(2) << sec << '.' <<
		setw(4) << milisec;
	
	return time.str();
}

bool isPrime(unsigned int num) {
	bool res = true;
	for(unsigned int i = 2; i < num; i++)
		if (num % i == 0) res = false;

	return res;
}

int main() {
	int primes = 0;
	clock_t t;
	string time;
	
	t = clock();
	for(unsigned int i = MIN; i <= MAX; i++)
		primes += isPrime(i);
	t = clock() - t;
	time = formatTime(((long double)t)/CLOCKS_PER_SEC);

	cout << "Total: " << primes << endl;
	cout << "Tiempo: " << time << endl;

	return 0;
}
